### PR TITLE
feat: CLI collection runner command generation UI flow

### PR DESCRIPTION
**Pull request by jamesgeorge007**
### Description

#4095 adds support for accessing collections/environments from a team workspace straightaway via the CLI by supplying the respective IDs without having to obtain an export. This PR includes UI flow updates to support the same where the CLI command to run a certain collection with the ID can be obtained (optionally including the active environment) via the context menu.

Closes HFE-520.

### Changes

- Fixes the case about dismissing the environment context menu (opened via pressing the `More` kebab menu) via pressing `Escape`, which results in an exception.
  
  image

- Made `editingProperties` a required prop for the `CollectionsProperties` component with `null` removed from the types since the existing usages guarantee the above prop to be an object.

> The below additions apply only to team workspaces.

- New `Run collection` action under the context menu option for collections that displays the modal with the respective CLI command to run the collection, optionally including any active environment.
  
  image

  image

- The `Collection Properties` modal now includes a new `Details` tab that shows the collection ID.
  
  image

- There'll be a new `properties` action under the context menu for environments that opens a new `Environment Properties` modal which displays the respective environment ID.

  image

  image

### CODE REVIEW

**Comment by jamesgeorge007** [comment]
> we can add a copy button close to the text, similar to the other modals.

Discussed separately, the copy to clipboard action next to the generated CLI command made it appear crowded and a primary copy action already exists.

**Review by nivedin — APPROVED** [review]

File: packages/hoppscotch-common/src/components/collections/Collection.vue [hunk 1]
```diff
@@ -267,16 +289,18 @@ const emit = defineEmits<{
...
 
-const tippyActions = ref<TippyComponent | null>(null)
-const requestAction = ref<HTMLButtonElement | null>(null)
-const folderAction = ref<HTMLButtonElement | null>(null)
-const edit = ref<HTMLButtonElement | null>(null)
-const deleteAction = ref<HTMLButtonElement | null>(null)
-const exportAction = ref<HTMLButtonElement | null>(null)
-const options = ref<TippyComponent | null>(null)
-const propertiesAction = ref<TippyComponent | null>(null)
+const tippyActions = ref<VNodeRef | null>(null)
```
Comment (amk-dev) [inline packages/hoppscotch-common/src/components/collections/Collection.vue:295]: what is the reason behind using `VNodeRef` here ? It includes very generic values like 

1. `Ref<any>` which will match any ref
2. string which will match any string

so typescript will not catch assignments like this,
```typescript
tippyActions.value = "whatever" // no TS error
``` 

look into the vue docs, on their recommendation for typing refs.
[https://vuejs.org/guide/typescript/composition-api#typing-template-refs](https://vuejs.org/guide/typescript/composition-api#typing-template-refs)

Comment (jamesgeorge007) [reply packages/hoppscotch-common/src/components/collections/Collection.vue]: Noticed a few type errors arising in the markup with a few property access via the template ref; picked the inferred type from the `ref` attribute. Yeah, better to narrow it down; push an [update](https://github.com/hoppscotch/hoppscotch/pull/4141/commits/9e1a454f8c76877b261a4a31bd35031cef34cebd).

**Review by amk-dev — APPROVED** [review]
LGTM ✅

**Pull request merged**